### PR TITLE
Rename prepare goal to generate-code

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
CC @stuartwdouglas @rsvoboda

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 1]
```diff
@@ -119,7 +119,7 @@
 
     private static final String QUARKUS_PLUGIN_GROUPID = "io.quarkus";
     private static final String QUARKUS_PLUGIN_ARTIFACTID = "quarkus-maven-plugin";
-    private static final String QUARKUS_PREPARE_GOAL = "prepare";
+    private static final String QUARKUS_PREPARE_GOAL = "generate-code";
```
Comment (gsmet) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:122]: The constant should probably be renamed too?

File: docs/src/main/asciidoc/grpc-getting-started.adoc [hunk 2]
```diff
@@ -37,7 +37,7 @@ Make sure you have `prepare` goal of `quarkus-maven-plugin` enabled in your `pom
             <executions>
                 <execution>
                     <goals>
-                        <goal>prepare</goal>
+                        <goal>generate-code</goal>
```
Comment (rsvoboda) [inline docs/src/main/asciidoc/grpc-getting-started.adoc:40]: Why isn't `<goal>generate-code-tests</goal>` added in the example too?

File: docs/src/main/asciidoc/grpc-getting-started.adoc [hunk 3]
```diff
@@ -37,7 +37,7 @@ Make sure you have `prepare` goal of `quarkus-maven-plugin` enabled in your `pom
             <executions>
```
Comment (rsvoboda) [inline docs/src/main/asciidoc/grpc-getting-started.adoc:37]: On line 28:
Make sure you have `prepare` goal of `quarkus-maven-plugin` enabled in your `pom.xml`:
 => 
Make sure you have `generate-code` goal of `quarkus-maven-plugin` enabled in your `pom.xml`:

Comment (rsvoboda) [reply docs/src/main/asciidoc/grpc-getting-started.adoc:37]: Alternatively: `generate-code` and `generate-code-tests` goals

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Few changes were requested and I have two additional questions

 - Why are there separate goals `generate-code` and `generate-code-tests` goals?
 - Why isn't `generate-code-tests` goal explained in grpc-getting-started.adoc?

File: docs/src/main/asciidoc/grpc-getting-started.adoc (hunk 2 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/grpc-getting-started.adoc:40]: :+1:

File: docs/src/main/asciidoc/grpc-getting-started.adoc (hunk 3 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/grpc-getting-started.adoc:38]: done

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java (hunk 1 above)
Comment (michalszynkiewicz) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: done

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**